hyper/http11/response.py

```diff
@@ -48,7 +49,8 @@ def __init__(self, code, reason, headers, sock, connection=None):
         # bother checking for content-length, we just keep reading until
         # we no longer can.
         self._expect_close = False
-        if b'close' in self.headers.get(b'connection', []):
+        if (b'close' in self.headers.get(b'connection', []) or
+                b'close' in self.headers.get(b'proxy-connection', [])):
             self._expect_close = True
```

Comment:
So do you suggest to ignore the `proxy-connection` header at all?

How should a response like this one be handled? It causes an assertion error if we ignore the `proxy-connection` header when using an insecure proxy (w/o CONNECT).

    HTTP/1.0 407 Proxy Authentication Required\r\n
    Proxy-Authenticate: Basic realm="proxy"\r\n
    Proxy-Connection: close\r\n
    \r\n